File: componentes/src/main/java/com/corporacao/capsula/componentes/fragment/BaseFragment.java
package com.corporacao.capsula.componentes.fragment;

import android.support.v4.app.Fragment;

import com.corporacao.capsula.componentes.util.ActionMaps;
import com.corporacao.capsula.componentes.util.BaseView;

import es.dmoral.toasty.Toasty;


public abstract class BaseFragment extends Fragment implements BaseView, ActionMaps {

    public abstract void createRestListener();
    public abstract void mapComponents();
    public abstract void mapComponentActions();

    @Override
    public void onResume() {
        super.onResume();
        if (this instanceof HasPresenter && ((HasPresenter) this).getPresenter() != null) {
            ((HasPresenter) this).getPresenter().resume();
        }
    }

    @Override
    public void onPause() {
        super.onPause();
        if (this instanceof HasPresenter && ((HasPresenter) this).getPresenter() != null) {
            ((HasPresenter) this).getPresenter().pause();
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (this instanceof HasPresenter && ((HasPresenter) this).getPresenter() != null) {
            ((HasPresenter) this).getPresenter().destroy();
        }
    }

    @Override
    public void showMessage(String message) {
        Toasty.success(getActivity(), message, Toasty.LENGTH_SHORT, true).show();
    }

    @Override
    public void showErrorMessage(String aMessage) {
        Toasty.error(getActivity(), aMessage, Toasty.LENGTH_SHORT, true).show();
    }

    @Override
    public void showWarningMessage(String aMessage) {
        Toasty.warning(getActivity(), aMessage, Toasty.LENGTH_SHORT, true).show();
    }

    @Override
    public void showWarningMessageLong(String aMessage) {
        Toasty.warning(getActivity(), aMessage, Toasty.LENGTH_LONG, true).show();
    }

    @Override
    public void showInfoMessage(String aMessage) {
        Toasty.info(getActivity(), aMessage, Toasty.LENGTH_SHORT, true).show();
    }
}

File: app/src/main/java/com/buffalo/controlefinancas/util/fragment/BaseFragment.kt
package com.buffalo.controlefinancas.util.fragment

import androidx.fragment.app.Fragment
import com.buffalo.controlefinancas.util.BaseView
import com.buffalo.controlefinancas.util.ActionMaps
import es.dmoral.toasty.Toasty

abstract class BaseFragment : Fragment(), BaseView, ActionMaps {
    abstract fun createRestListener()
    abstract override fun mapComponents()
    abstract override fun mapComponentActions()
    override fun onResume() {
        super.onResume()
        if (this is HasPresenter<*> && (this as HasPresenter<*>).presenter != null) {
            (this as HasPresenter<*>).presenter.resume()
        }
    }

    override fun onPause() {
        super.onPause()
        if (this is HasPresenter<*> && (this as HasPresenter<*>).presenter != null) {
            (this as HasPresenter<*>).presenter.pause()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        if (this is HasPresenter<*> && (this as HasPresenter<*>).presenter != null) {
            (this as HasPresenter<*>).presenter.destroy()
        }
    }

    override fun showMessage(message: String) {
        Toasty.success(requireActivity(), message, Toasty.LENGTH_SHORT, true).show()
    }

    override fun showErrorMessage(aMessage: String) {
        Toasty.error(requireActivity(), aMessage, Toasty.LENGTH_SHORT, true).show()
    }

    override fun showWarningMessage(aMessage: String) {
        Toasty.warning(requireActivity(), aMessage, Toasty.LENGTH_SHORT, true).show()
    }

    override fun showWarningMessageLong(aMessage: String) {
        Toasty.warning(requireActivity(), aMessage, Toasty.LENGTH_LONG, true).show()
    }

    override fun showInfoMessage(aMessage: String) {
        Toasty.info(requireActivity(), aMessage, Toasty.LENGTH_SHORT, true).show()
    }
}